error.go

```diff
@@ -117,10 +118,22 @@ func wrap(err error) (wrappedErr *Error) {
 // Unwrap extracts an underlying *fail.Error from an error.
 // If the given error isn't eligible for retriving context from,
 // it returns nil
-func Unwrap(err error) *Error {
+func Unwrap(err error) (unwrapped *Error) {
+	if err == nil {
+		return nil
+	}
+
 	if appErr, ok := err.(*Error); ok {
 		return appErr
 	}
 
+	pkgErr := extractPkgError(err)
+	if appErr, ok := pkgErr.Err.(*Error); ok {
```

Comment:
Wait. Maybe `pkgErr.Err` doesn't always have to be `*fail.Error`.
Let me think again tmrw with my fresh eyes..